### PR TITLE
Throw exception on invalid calls

### PR DESCRIPTION
**Pull request by Flamefire**
This adds exception throwing for the invalid methods of ParallelDC.

I just run into an issue accidentally using one of them, which is pretty annoying.

### CODE REVIEW

File: src/ParallelDataCollector.cpp [hunk 1]
```diff
@@ -770,33 +770,6 @@ namespace splash
...
-
-        if (srcID != dstID)
-            throw DCException(getExceptionString("createReference",
-                "source and destination ID must be identical", NULL));
-
-        if (srcName == dstName)
-            throw DCException(getExceptionString("createReference",
-                "a reference must not be identical to the referenced data", srcName));
-
-        throw DCException(getExceptionString("createReference",
```
Comment (ax3l) [inline src/ParallelDataCollector.cpp]: I like the twist at the end! :guitar:

File: src/ParallelDataCollector.cpp [hunk 2]
```diff
@@ -1139,4 +1112,56 @@ namespace splash
         dataset.close();
     }
 
+    /* UNIMPLEMENTED METHODS FROM DATACOLLECTOR. TODO: Unify interface to remove those */
+    void ParallelDataCollector::readGlobalAttribute(const char*, void*, Dimensions*)
```
Comment (ax3l) [inline src/ParallelDataCollector.cpp]: I don't get those: PDC *should* already have a usable `readGlobalAttribute` method, no?

Comment (Flamefire) [inline src/ParallelDataCollector.cpp]: Yes, but with different signatures

Comment (ax3l) [inline src/ParallelDataCollector.cpp]: it's just those specific signatures it does not have?

**Comment by ax3l** [comment]
thank you!

File: src/ParallelDataCollector.cpp (hunk 2 above)
Comment (Flamefire) [inline src/ParallelDataCollector.cpp]: Yes. This are the ones, that were not implemented before (empty, see the header: https://github.com/ComputationalRadiationPhysics/libSplash/pull/247/files/2bb3e900bb3aa67a2ef741ff3b2e98835e21ecd2#diff-786f550b56fdbb1aec6013d83cd5085fL360)

**Pull request merged**